test/unit/Schema.js

```diff
@@ -1156,6 +1155,57 @@ describe("Schema", () => {
 		});
 	});
 
+	describe("getIndexAttributes", () => {
+		const tests = [
+			// Defaults
+			{
+				"name": "Should return an empty array if no indicies are defined",
+				"schema": {"id": String},
+				"output": []
+			},
+			{
+				"name": "Should return an array containing definitions for an boolean defined index",
+				"schema": {"id": {"type": String, "index": true}},
+				"output": [{"attribute": "id", "index": true}]
+			},
+			{
+				"name": "Should return an array containing definitions for an object defined index",
+				"schema": {"id": {"type": String, "index": {"global": true, "name": "id-index"}}},
+				"output": [{"attribute": "id", "index": {"global": true, "name": "id-index"}}]
+			},
+			{
+				"name": "Should return an array containing definitions for an array defined index",
+				"schema": {"id": {"type": String, "index": [{"global": true, "name": "id-index"}]}},
+				"output": [{"attribute": "id", "index": {"global": true, "name": "id-index"}}]
+			},
+			{
+				"name": "Should return an array containing multiple definitions for array defined indexes",
+				"schema": {"id": {"type": String, "index": [{"global": true, "name": "id-index-1"}, {"global": false, "name": "id-index-2"}]}},
+				"output": [{"attribute": "id", "index": {"global": true, "name": "id-index-1"}}, {"attribute": "id", "index": {"global": false, "name": "id-index-2"}}]
+			},
+			{
+				"name": "Should aggregate an array containing multiple definitions for all defined indexes",
+				"schema": {"id": {"type": String, "index": [{"global": true, "name": "id-index-1"}, {"global": false, "name": "id-index-2"}]}, "uid": {"type": String, "index": {"global": true, "name": "uid-index"}}, "uuid": {"type": String}},
+				"output": [{"attribute": "id", "index": {"global": true, "name": "id-index-1"}}, {"attribute": "id", "index": {"global": false, "name": "id-index-2"}}, {"attribute": "uid", "index": {"global": true, "name": "uid-index"}}]
+			}
+		];
+
+		tests.forEach((test) => {
+			it(test.name, async () => {
+				const schema = new dynamoose.Schema(test.schema);
+				const output = await schema.getIndexAttributes();
+				if (typeof output !== "function") {
+					expect(output).to.eql(test.output);
+				} else {
+					expect(typeof output).to.eql(typeof test.output);
+					expect(output.toString()).to.eql(test.output.toString());
+					expect(await output()).to.eql(await test.output());
+				}
```

Comment:
What is the purpose of this else block? It looks to me like it'll never get run. Is there a longer term vision here?

Reply:
No, just copy & paste.

Resolved in https://github.com/dynamoose/dynamoose/pull/1079/commits/ca9f42593e058b16b7ba2246b91ff42acf6bc16e